api/src/test/java/com/epam/pipeline/manager/pipeline/FolderTemplateManagerTest.java

```diff
@@ -75,98 +67,130 @@
+        Assert.assertEquals(metadata, folderTemplate.getMetadata());
+        Assert.assertEquals(1, folderTemplate.getPermissions().size());
 
-        AclTestDao.AclClass folderAclClass = new AclTestDao.AclClass(Folder.class.getCanonicalName());
-        aclTestDao.createAclClassIfNotPresent(folderAclClass);
+        Assert.assertEquals(permissionVO, folderTemplate.getPermissions().get(0));
+
+        postValidation();
     }
 
-    @Test
-    @WithMockUser(username = TEST_USER)
-    public void createFolderFromTemplateTest() throws IOException {
-        Map<String, PipeConfValue> metadata = new HashMap<>();
-        metadata.put(DATA_KEY_1, new PipeConfValue(DATA_TYPE_1, DATA_VALUE_1));
+    private void initializeParameters() {
+        metadata = getMetadata();
 
         DataStorageWithMetadataVO dataStorageVO = new DataStorageWithMetadataVO();
         dataStorageVO.setName(DATASTORAGE_NAME_1);
         dataStorageVO.setType(DataStorageType.S3);
         dataStorageVO.setPath(TEST_PATH);
         dataStorageVO.setMetadata(metadata);
 
-        PermissionVO permissionVO = new PermissionVO();
+        permissionVO = new PermissionVO();
         permissionVO.setMask(AclPermission.READ.getMask());
         permissionVO.setUserName(TEST_ROLE);
         permissionVO.setPrincipal(false);
 
-        FolderTemplate childFolderTemplate1 = FolderTemplate.builder().name(CHILD_TEMPLATE_FOLDER_NAME_1).build();
-        FolderTemplate folderTemplate = FolderTemplate.builder()
+        FolderTemplate childFolderTemplate1 = FolderTemplate.builder()
+                .name(CHILD_TEMPLATE_FOLDER_NAME_1)
+                .datastorages(Stream.of(new DataStorageWithMetadataVO()).collect(Collectors.toList()))
+                .build();
+
+        folderTemplate = FolderTemplate.builder()
                 .name(TEMPLATE_FOLDER_NAME)
                 .datastorages(Stream.of(dataStorageVO).collect(Collectors.toList()))
                 .children(Stream.of(childFolderTemplate1).collect(Collectors.toList()))
                 .metadata(metadata)
                 .permissions(Stream.of(permissionVO).collect(Collectors.toList()))
                 .build();
+    }
+
+    private void preConditions() {
+        doAnswer(invocation -> {
+            Folder parameterFolder = invocation.getArgumentAt(0, Folder.class);
+            return (FOLDER_ID.equals(parameterFolder.getParentId())) ? getFolderChild() : getFolder();
+        }).when(mockFolderCrudManager).create(any(Folder.class));
+
+        doReturn(getMockSecurityEntity()).when(mockDataStorageManager)
+                .create(any(DataStorageVO.class), anyBoolean(), anyBoolean(), anyBoolean());
+        doNothing().when(mockMetadataManager).updateEntityMetadata(anyMapOf(String.class, PipeConfValue.class),
+                anyLong(), any(AclClass.class));
+        doNothing().when(spyGrantPermissionManager).setPermissionsToEntity(anyListOf(PermissionVO.class),
```

Comment:
I would suggest not mocking the spy. Especially with such a common matchers. It can possibly break tests in other tests classes with the same spring context.

What you have to do instead is create an acl entity using `AbstractAclTest.initAclEntity`.

Reply:
Fixed